one_off_scripts/backfill_test_flag_bridges.py

```diff
@@ -0,0 +1,54 @@
+import logging
+
+from django.db import transaction as django_transaction
+from shared.django_apps.core.models import Repository
+from shared.django_apps.reports.models import (
+    RepositoryFlag,
+    Test,
+    TestFlagBridge,
+    TestInstance,
+)
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger()
+
+
+def backfill_test_flag_bridges(repoid=None):
+    log.info("Backfilling TestFlagBridge objects", extra=dict(repoid=repoid))
+    repos = Repository.objects.filter(test_analytics_enabled=True)
+    if repoid is not None:
+        repos = repos.filter(repoid=repoid)
+
+    for repo in repos:
+        tests = Test.objects.filter(repository_id=repo.repoid)
+
+        flags = {
+            flag.flag_name: flag
+            for flag in RepositoryFlag.objects.filter(repository=repo)
+        }
+
+        bridges_to_create = []
+        for test in tests:
+            TestFlagBridge.objects.filter(test=test).delete()
+
+            try:
+                first_test_instance = (
+                    TestInstance.objects.filter(test_id=test.id)
+                    .select_related("upload")
+                    .first()
```

Comment:
my comment might have been confusing. I believe `first` returns `None` if the element does not exist instead of the `IndexError`.